packages/blade/src/components/Tour/_decisions/decisions.md

```diff
@@ -0,0 +1,723 @@
+# Tour API Decisions <!-- omit in toc -->
+
+The Tour component is used to provide context as well as enable users to take certain actions on it. These are used to highlight a new feature or provide a guided tour to a new user.
+
+<img src="./assets/tour-thumbnail.png" width="380" alt="Tooltip Thumbnail" />
+
+- [Design](#design)
+- [Features](#features)
+- [API](#api)
+  - [`Tour` API](#tour-api)
+  - [`TourStep` API](#tourstep-api)
+- [Usage](#usage)
+- [Motion](#motion)
+- [Open Questions And Technical Challenges](#open-questions-and-technical-challenges)
+- [React Native Specifics](#react-native-specifics)
+- [Multiple Tour Flows](#multiple-tour-flows)
+  - [Approach-1: Prefix `id` to avoid conflicts](#approach-1-prefix-id-to-avoid-conflicts)
+  - [Approach-2: Use `TourStep` to avoid conflicts](#approach-2-use-tourstep-to-avoid-conflicts)
+  - [Technical Challenge in React Native](#technical-challenge-in-react-native)
+- [API Design Challenges](#api-design-challenges)
+  - [Approach: Let Consumer Compose `Popover`](#approach-let-consumer-compose-popover)
+    - [**1. Not possible to collocate the tour flows.**](#1-not-possible-to-collocate-the-tour-flows)
+    - [**2. How will we have two different tour flows in the same page?**](#2-how-will-we-have-two-different-tour-flows-in-the-same-page)
+    - [3. Segmentation / Lack of control](#3-segmentation--lack-of-control)
+    - [4. Not possible to maintain a consistent API / Implementation between web \& native](#4-not-possible-to-maintain-a-consistent-api--implementation-between-web--native)
+    - [Conclusion](#conclusion)
+- [MoMs](#moms)
+- [References](#references)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=40540-559304&t=tmTrf3xJU6oj59fM-0) to all variants of the Popover component
+
+## Features
+
+- [x] **Tour:** A guided tour with multiple steps
+- [x] **Masking:** Mask the rest of the page except the highlighted element.
+
+## API
+
+### `Tour` API
+
+```jsx
+// Step will have similar props as the Popover component, With extra Tour related props.
+type Step = {
+  /**
+   * Unique identifier for the tour step
```

Comment:
Maybe just mention identifier? the moment we say unique it tells me that i need to use some random number or uuid to have this identifier?

Reply:
We should mention "unique", since it has to be unique, you cannot have duplicate identifiers in the steps array.